### PR TITLE
blackbox-exporter CRD

### PR DESCRIPTION
**Pull request by paulfantom**

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	// Secure listening port
+	Port int `json:"port,omitempty"`
+	// Insecure listening port
+	InternalPort int `json:"internalPort,omitempty"` // TODO: we should probably remove this
+	// Number of replicas of blackbox-exporter
+	Replicas int `json:"replicas,omitempty"`
+	// Configuration of blackbox-exporter in the same form as in
+	Modules map[string]string `json:"modules,omitempty"` // TODO: Consider using blackbox-exporter types for this
+	// Labels applied to all blackbox-exporter objects
+	CommonLabels map[string]string `json:"commonLabels,omitempty"`
```
Comment (brancz) [inline pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go]: I feel like the hidden fields should probably not be exposed here. If people want to drop to this level, they should use jsonnet I think.

Comment (paulfantom) [reply pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go]: agreed, I'll remove it

File: pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	Version string `json:"version,omitempty"`
+	// Image of config-reloader sidecar container
+	// +kubebuilder:default="jimmidyson/configmap-reload:v0.5.0"
+	ConfigmapReloaderImage string `json:"configmapReloaderImage,omitempty"`
+	// Secure listening port
+	// +kubebuilder:default=19115
+	Port int `json:"port,omitempty"`
+	// Insecure listening port
+	// +kubebuilder:default=9115
+	InternalPort int `json:"internalPort,omitempty"` // TODO: we should probably remove this
```
Comment (metalmatze) [inline pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go]: Then let's remove it now and wait for people to explicitly ask for it and the reason why. :)

**Comment by metalmatze** [comment]
nice!

File: pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go (hunk 2 above)
Comment (paulfantom) [reply pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go]: Makes sense. I removed it. :)

**Comment by paulfantom** [comment]
I think this one is done. I am starting to implement the same for KSM.

**Comment by brancz** [comment]
as this is all experiments, feel free to merge your own PRs

**Pull request merged**